src/ModelManager.ts

```diff
@@ -93,90 +93,147 @@ export class ModelManager {
      * Once the initial model is loaded and if the data model doesn't contain the path of the current pathname, the library attempts to fetch a fragment of model.
      *
      * @param {string|InitializationConfig} [config]                - URL to the data model or configuration object
-     * @fires cq-pagemodel-loaded
      * @return {Promise}
      */
     public initialize<M extends Model>(config?: ModelManagerConfiguration | string): Promise<M> {
+        this.initializeAsync(config);
+        return this._initPromise;
+    }
+
+    /**
+     * Initializes the ModelManager asynchronously using the rootModelPath to resolve a data model.
+     * For remote apps with no defined root model, an empty store is initialized and data is fetched on demand by components.
+     *
+     * Once the initial model is loaded and if the data model doesn't contain the path of the current pathname,
+     * the library attempts to fetch a fragment of model.
+     *
+     * @param {string|InitializationConfig} [config]                - URL to the data model or configuration object
+     * @fires cq-pagemodel-loaded if root model path is available
+     */
+    public initializeAsync<M extends Model>(config?: ModelManagerConfiguration | string): void {
         this.destroy();
-        let path;
-        let initialModel = null;
 
-        if (!config || (typeof config === 'string')) {
-            path = config;
-        } else if (config) {
-            path = config.path;
-            this._modelClient = config.modelClient;
-            initialModel = config.model;
+        const modelConfig = this._toModelConfig(config);
+        this._initializeFields(modelConfig);
+
+        const initialModel = modelConfig && modelConfig.model;
+        const {
+            rootModelPath,
+            rootModelURL,
+            currentPathname,
+            sanitizedCurrentPathname,
+            metaPropertyModelUrl
+        } = this._getPathsForModel(modelConfig);
+
+        this._modelStore = new ModelStore(rootModelPath, initialModel);
+
+        if (rootModelPath) {
```

Comment:
Quite a big block of nested if-else. Maybe we can reduce the cognitive complexity here.